perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_group_attribute_def_virt_adDisplayName_o365muTest.java

```diff
@@ -0,0 +1,153 @@
+	private final AttributeDefinition inetGroupNameAbbENAttrDef = setUpGroupAttributeDefinition(4, "inetGroupNameAbbEN", String.class.getName());
+	private final AttributeDefinition inetWorkplacesTypeCSAttrDef = setUpGroupAttributeDefinition(5, "inetWorkplacesTypeCS", String.class.getName());
+
+	private final String typeOfWorkplacesA = "typeA";
+	private final String typeOfWorkplacesB = "typeB";
+	private final String typeOfWorkplacesC = "typeC";
+	private final String predefinedDisplayName = "specialTestingPredefinedDisplayName";
+
+	private PerunSessionImpl sess;
+
+	@Before
+	public void setUp() throws Exception {
+		//prepare mocks
+		sess = mock(PerunSessionImpl.class);
+		PerunBl perunBl = mock(PerunBl.class);
+		AttributesManagerBl am = mock(AttributesManagerBl.class);
+		GroupsManagerBl gm = mock(GroupsManagerBl.class);
+		when(sess.getPerunBl()).thenReturn(perunBl);
+		when(perunBl.getAttributesManagerBl()).thenReturn(am);
+		when(perunBl.getGroupsManagerBl()).thenReturn(gm);
+
+		when(sess.getPerunBl().getGroupsManagerBl().getParentGroup(sess, groupB)).thenReturn(groupA);
+		when(sess.getPerunBl().getGroupsManagerBl().getParentGroup(sess, groupC)).thenReturn(groupB);
+		when(sess.getPerunBl().getGroupsManagerBl().getParentGroup(sess, groupD)).thenReturn(groupC);
+		when(sess.getPerunBl().getGroupsManagerBl().getParentGroup(sess, groupE)).thenReturn(groupD);
+
+		//group A is the main group with cispr 000000, it has no default display name
+		setWhenClauseByParametersForGroup(sess, groupA, null, "000000", groupA.getName(), groupA.getDescription(), typeOfWorkplacesA);
+		//group B is special group which should return null on display name
+		setWhenClauseByParametersForGroup(sess, groupB, null, "000001", groupB.getName(), groupB.getDescription(), typeOfWorkplacesA);
+		//group C is other part of university, it has no default display name
+		setWhenClauseByParametersForGroup(sess, groupC, null, "110000", groupC.getName(), groupC.getDescription(), typeOfWorkplacesB);
+		//group D is special group with predefined display name
+		setWhenClauseByParametersForGroup(sess, groupD, predefinedDisplayName, null, null, null, null);
+		//group E is other type of group, it has no default display name
+		setWhenClauseByParametersForGroup(sess, groupE, null, "123456", groupE.getName(), groupE.getDescription(), typeOfWorkplacesC);
+
+		when(sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, groupWithoutAttributes, defaultAdDisplayNameAttrDef.getName())).thenThrow(AttributeNotExistsException.class);
+		when(sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, groupWithoutAttributes, inetCisprAttrDef.getName())).thenThrow(AttributeNotExistsException.class);
+		when(sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, groupWithoutAttributes, inetWorkplacesTypeCSAttrDef.getName())).thenThrow(AttributeNotExistsException.class);
```

Comment:
It would be nice to have the `when` and `.then` calls on separate lines, so it is easier to distinguish between those two parts.

Reply:
Changed.